tests/upgrades/test_client.py

```diff
@@ -27,373 +28,303 @@
-            ak_name=self.ak_name,
-            client_os='rhel7',
-            org=self.org,
-            environment=environment,
-            sat_state='pre',
-        )
-        rhel7_client = dockerize(ak_name=ak.name, distro='rhel7', org_label=self.org.label)
-        client_container_id = list(rhel7_client.values())[0]
-        product, yum_repo = create_yum_test_repo(
-            product_name=self.prod_name, repo_url=settings.repos.fake_repo_zoo3, org=self.org
-        )
-        update_product_subscription_in_ak(product=product, yum_repo=yum_repo, ak=ak, org=self.org)
-        attach_custom_product_subscription(
-            prod_name=product.name, host_name=str(list(rhel7_client.keys())[0]).lower()
-        )
-        # Refresh subscriptions on client
-        execute(
-            docker_execute_command,
-            client_container_id,
-            'subscription-manager refresh',
-            host=self.docker_vm,
-        )
-        # Run goferd on client as its docker container
-        kwargs = {'async': True, 'host': self.docker_vm}
-        execute(docker_execute_command, client_container_id, 'goferd -f', **kwargs)
-        status = execute(
-            docker_execute_command, client_container_id, 'ps -aux', host=self.docker_vm
-        )[self.docker_vm]
-        self.assertIn('goferd', status)
-
-        create_dict({self.__class__.__name__: rhel7_client})
-
-    @post_upgrade(depend_on=test_pre_scenario_preclient_package_installation)
-    def test_post_scenario_preclient_package_installation(self):
-        """Post-upgrade scenario that installs the package on pre-upgrade
-        client remotely and then verifies if the package installed
-
-        :id: postupgrade-eedab638-fdc9-41fa-bc81-75dd2790f7be
-
-        :steps: Install package on a pre-upgrade client
-
-        :expectedresults: The package is installed in client
-        """
-        client = get_entity_data(self.__class__.__name__)
-        client_name = str(list(client.keys())[0]).lower()
-        client_id = entities.Host().search(query={'search': f'name={client_name}'})[0].id
-        entities.Host().install_content(
-            data={
-                'organization_id': self.org.id,
-                'included': {'ids': [client_id]},
-                'content_type': 'package',
-                'content': [self.package_name],
-            }
-        )
-        # Validate if that package is really installed
-        installed_package = execute(
-            docker_execute_command,
-            list(client.values())[0],
-            f'rpm -q {self.package_name}',
-            host=self.docker_vm,
-        )[self.docker_vm]
-        self.assertIn(self.package_name, installed_package)
-
-
-class TestScenarioUpgradeNewClientAndPackageInstallation(APITestCase):
-    """The test class contains post-upgrade scenarios to test if the package
+
+@pre_upgrade
+def test_pre_scenario_preclient_package_installation(
```

Comment:
The `@pre_upgrade` and `@post_upgrade` must be the part class context and not module. The only reason behind that is to upload the 2 defs(pre and post) as one customer scenario to polarion .

Reply:
Class binds them together !

Reply:
I have now put classes back.